### PR TITLE
Doc: Add graph of call made by distributed

### PR DESCRIPTION
**Pull request by kinnou02**
Naming isn't probably very clear, so if you have better ideas...

### CODE REVIEW

**Review by nonifier — APPROVED** [review]

File: documentation/diagrams/distributed.gv [hunk 1]
```diff
@@ -0,0 +1,37 @@
+digraph distributed {
```
Comment (pbougue) [inline documentation/diagrams/distributed.gv:4]: could be nice to add a comment on URL of request in the file

File: documentation/diagrams/distributed.gv [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "destination free access" -> "PT walking - walking"
+
+    //PT bike - walking
+    "bike origin sn matrix" -> "PT bike - walking"
+    "walking destination sn matrix" -> "PT bike - walking"
+    "bike direct path" -> "PT bike - walking"
+    "origin free access" -> "PT bike - walking"
+    "destination free access" -> "PT bike - walking"
+
+    "PT bike - walking" -> "bike origin fallback"  -> "build bike-walking" -> "final response"
```
Comment (pbougue) [inline documentation/diagrams/distributed.gv]: I'd rename those "PT bike - walking" to "RAPTOR bike>PT>walking" (same for the other one)

File: documentation/diagrams/distributed.gv (hunk 1 above)
Comment (pbougue) [inline documentation/diagrams/distributed.gv:4]: Maybe adding a specific color/shape for kraken calls would ease reading (hope I didn't miss any), and add a comment in doc saying that kraken is the square-blue thing
```
    "resolve origin" [shape=box, color=lightblue, style=filled];
    "resolve destination" [shape=box, color=lightblue, style=filled];
    "PT bike - walking" [shape=box, color=lightblue, style=filled];
    "PT walking - walking" [shape=box, color=lightblue, style=filled];
```

File: documentation/rfc/distributed.md [hunk 3]
```diff
@@ -39,3 +39,9 @@ from the origin to the departure stop_point of the first section and another jou
 stop_point of the last section to the destination.
 Like others operations this is done asynchronously as soon as we get a kraken response and if multiple journeys
 use the same fallback journey, it will only be computed one time.
+
+By example for a request like this one: https://api.navitia.io/v1/coverage/fr-idf/journeys?from=stop_area%3AOIF%3ASA%3A59238&to=2.33071%3B48.83890&first_section_mode%5B%5D=bike&first_section_mode%5B%5D=walking&datetime=20180614T113500&last_section_mode%5B%5D=walking&
+
+We will do this:
```
Comment (pbougue) [inline documentation/rfc/distributed.md]: I'd precise that arrows are dependencies

File: documentation/diagrams/distributed.gv [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    "PT bike - walking" -> "walking destination fallback" -> "build bike-walking"
+
+    "PT walking - walking" -> "walking origin fallback" -> "build walking-walking" -> "final response"
```
Comment (pbougue) [inline documentation/diagrams/distributed.gv]: I'd rename "build walking-walking" to "build responses walking>PT>walking" (same for the other one)

**Review by pbougue — COMMENTED** [review]
Looks good!

**Comment by benoit-bst** [comment]
A sequence diagram is not better ?
*Example* :
https://cote.io/2014/07/10/a-sequence-diagram/

**Comment by benoit-bst** [comment]
Link for a sequence diagram generator :
https://sequencediagram.org/

**Comment by mbounabi** [comment]
As discussed, a sequence diagram will be better. To be done later?

File: documentation/diagrams/distributed.gv (hunk 2 above)
Comment (mbounabi) [reply documentation/diagrams/distributed.gv]: Is it necessary to mention RAPTOR here?

**Comment by mbounabi** [comment]
After discussion with @TeXitoi @xlqian @benoit-bst , it seems that the sequence diagram will be too long and maybe not easier to read.
In order to improve the readability of the diagram, we will need to:
* add an entrypoint (the journey request done before resolve origin/destination)
* optional: origin related actions on the left side of the diagram and destiantion on the right
* a way to tell the difference between the streenetwork and PT actions (different form, color)
* Separate the diagram in 3 distinct steps:
	* actions done to feed the Kraken for PT
	* PT calculation
	* Journey assembly with PT and fallbacks

**Comment by nonifier** [comment]
what about something like this ? [distributed sequencediagram](https://sequencediagram.org/index.html#initialData=C4S2BsFMAIBEQM7AE4gEYFdiQCYCg8BDAY2AHtloAlSARw0iQJvseAFoA+AKQoFsMAOwBcAegBWZDMkGQAngUFls0VAHMAFsGhkAZtF7IBImgjLgAbjEiCUc6AAcyIWwmhFSIC4RWHj0AGI1ZEIFcBdIQmQ8PyEAHi4AaRCAaxthAAVwEkgAITkAVVRoAG0KEDUXABpoHDYXHxAyQQBdPGTCNMEEzg6u4Qds4kgAfWkQPHDZKJ1dXTwEBxy8OpJQb19+IUVlGHUtWYMtkXhkSFJHHw1oXQpoSCtkez4yOo91nxhYwUDg0Jjjgkkql0gB1QjgFIuNTQdzfIGcABqEI0EOywgAwlEAUZ4uwuABxSBkKzgMjCXIgNILJbDdyrTwbL7HHYqfbaPRHXEiACCxGGCAQ6Cg0AAyuQHI5nK53GsvJ8uf4giEFPDgZ10oMcggRtNkGh7GV1NVavVBI1mm0ppFKDjjAi+ukstqAGJnSB8gUIdog7r43q+4SvYAjJBkBwjJwuYDe60zPTzBkfTbcgiLHLuPBKNkVA6c76Ysh8BxYGC6NFoEgpWrSC2Cb1ypmKoS-FWTCIzO149X9cGQ6HQF0VqtwWugZrex1+nvpJDukOyYAAdwoKRGyCkoEEahGfB8qAAHl3p0iUWjCJiZkPwOBK8Rq7Ax0163hkeBUTfCAi3x-0XPIJAC6ASuyBrhuWDQru+4gEeaqEsSpLkpSaSDsO96jiE44vkSJKQGSCI4Yhwj-oBurAau66bpBe4oDB7Z6rMiaRIyCrfKyey5hy+gFhiRYlioGQYGg4TENAAAqIT1k4yDaJI0iyHIDYsSmSp-GEHa2mqAYaiIGTALw8nyD6Ok9FOAwhlqgiyNEcaUAmKzMcmzKpngLAMEgCQFnJMjyAgQA)

File: documentation/diagrams/distributed.gv (hunk 2 above)
Comment (nonifier) [reply documentation/diagrams/distributed.gv]: I agree with @mbounabi, RAPTOR is too low-level to be mentioned on a high-level documentation.

Comment (pbougue) [reply documentation/diagrams/distributed.gv]: Yep, it was to mention that we assemble things in there (PT and fallbacks) but it's just the maths computation, the result is still missing details (fallback complete paths and all)

**Review by mbounabi — APPROVED** [review]

**Review by pbougue — APPROVED** [review]

**Review by xlqian — APPROVED** [review]

**Pull request merged**